src/components/Tokens/TokenDetails/BalanceSummary.tsx

```diff
@@ -43,13 +45,40 @@ const TotalBalanceItem = styled.div`
   display: flex;
 `
 
+const BalanceRowLink = styled(StyledInternalLink)`
+  color: unset;
+`
+
+function BalanceRow({ currency, formattedBalance, formattedUSDValue, href }: BalanceRowData) {
+  const content = (
+    <TotalBalance key={currency.wrapped.address}>
+      <TotalBalanceItem>
+        <CurrencyLogo currency={currency} />
+        &nbsp;{formattedBalance} {currency?.symbol}
+      </TotalBalanceItem>
+      <TotalBalanceItem>{formatDollarAmount(formattedUSDValue === 0 ? undefined : formattedUSDValue)}</TotalBalanceItem>
+    </TotalBalance>
+  )
+  if (href) {
+    return <BalanceRowLink to={href}>{content}</BalanceRowLink>
+  }
+  return content
+}
+
+interface BalanceRowData {
+  currency: Currency
+  formattedBalance: number
+  formattedUSDValue: number | undefined
+  href?: string
+}
 export interface BalanceSummaryProps {
   tokenAmount: CurrencyAmount<Token> | undefined
   nativeCurrencyAmount: CurrencyAmount<Currency> | undefined
   isNative: boolean
 }
 
 export default function BalanceSummary({ tokenAmount, nativeCurrencyAmount, isNative }: BalanceSummaryProps) {
+  const pageChainName = useGlobalChainName()
```

Comment:
our page can be on a different page than the current global chain; probably want to grab the url param instead

Reply:
Ah good catch. Thought that was what this hook did because it is in `graphql/data/util`